src/components/Documentation/Layout/SidebarMenu/index.tsx

```diff
@@ -50,11 +50,31 @@ const SidebarMenuItem: React.FC<ISidebarMenuItemProps> = ({
   icon,
   type
 }) => {
-  const isActive = activePaths && includes(activePaths, path)
+  const [isActive, setIsActive] = useState(
+    activePaths && includes(activePaths, path)
+  )
+
+  useEffect(() => {
+    setIsActive(activePaths && includes(activePaths, path))
+  }, [activePaths])
+
   const isRootParent =
     activePaths && activePaths.length > 1 && activePaths[0] === path
   const isLeafItem = children === undefined || children.length === 0
-  const currentLevelOnClick = (): void => onClick(isLeafItem)
+  const currentLevelOnClick = (
+    event: SyntheticEvent<HTMLAnchorElement>
+  ): void => {
+    if (event.currentTarget.getAttribute('aria-current') === 'page') {
```

Comment:
A separate `isActive` with the old behavior could be useful here.

Reply:
Awesome, didn't thought about that, thank you.